### ISSUE TITLE
bug: component.d.ts generation gives TS errors

### ISSUE THREAD

**Issue by josip-kovacevic-valcon** [ionic-team/stencil#4999]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.6.0

### Current Behavior

We have simple `setup` pipeline which generates everything project needs. One of the files is `components.d.ts` file which is auto-generated by stencil. We git-ignored this file and also we made slim config file only for this file to be generated and on v4.5.0 and below it works nicely, but on v4.6.0 it gives error (look at picture).

Script used for dts generation: `"build.dts": "stencil build --config stencil-dts.config.ts"`
`stencil-dts.config.ts`
```ts
import { Config } from '@stencil/core';

export const config: Config = {
  namespace: 'vnp-web-components',
  sourceMap: false,
  transformAliasedImportPaths: true,
  outputTargets: [
    {
      type: 'dist-custom-elements',
    },
  ],
};

```

![image](https://github.com/ionic-team/stencil/assets/80685558/9c296b26-c175-42a6-bcd1-e1f9a5cb9e42)

### Expected Behavior

It shouldn't give errors. If there is simpler way only to generate `component.d.ts`, please provide it. I couldn't find anything in documentation nor I found similar issue.

### System Info

```shell
System: node 20.9.0
Platform: linux (6.2.0-35-generic)
CPU Model: 12th Gen Intel(R) Core(TM) i7-1265U (12 cpus)
Compiler: /home/josip-kovacevic/public-www/vnp-web-components/node_modules/@stencil/core/compiler/stencil.js
Build: 1698073283
Stencil: 4.6.0 💥
TypeScript: 5.2.2
Rollup: 2.42.3
Parse5: 7.1.2
Sizzle: 2.42.3
Terser: 5.22.0
```

### Steps to Reproduce

Make clean project setup using stencil cli. Try to use `build` or `start` script. It will give you TS errors since initially `components.d.ts` file does not exist.

### Code Reproduction URL

-

**Comment by verena-ifx** [comment]
same error for me, when trying to build via github-actions. Works fine up to v4.50., but throws the same error when trying with a higher version

**Comment by alicewriteswrongs** [comment]
Hey @josip-kovacevic-valcon, happy to report that the fix for this issue was released as part of today's [Stencil v4.7.1 release](https://github.com/ionic-team/stencil/releases/tag/v4.7.1)!
